extensions/mssql/src/sqlToolsServer.ts

```diff
@@ -87,11 +87,15 @@ export class SqlToolsServer {
 	}
 
 	private activateFeatures(context: AppContext): Promise<void> {
-		const credsStore = new CredentialStore(context.extensionContext.logPath, this.config);
 		const resourceProvider = new AzureResourceProvider(context.extensionContext.logPath, this.config);
-		this.disposables.push(credsStore);
 		this.disposables.push(resourceProvider);
-		return Promise.all([credsStore.start(), resourceProvider.start()]).then();
+		if (nativeCredentialsEnabled()) {
+			return Promise.all([resourceProvider.start()]).then();
```

Comment:
do we need `Promise.all` here?  it looks like there's just 1 promise so we can wait on it directly.